### ISSUE TITLE
builtin callback causes Graal to crash

### ISSUE THREAD

**Issue by mwaldrich** [Haiyang-Sun/nodeprof.js#65]
The `builtinEnter` and `builtinExit` callbacks sometimes provide values for `dis` that, when used in certain ways, crash Graal.

### Example program

```js
let p = new Promise((resolve, reject) => resolve(5));

p.then(v => console.log(v));

console.log("*** first event loop over");
```

### Example analysis
```js
// DO NOT INSTRUMENT
(function (sandbox) {
    function MyAnalysis() {
        this.isInterestingBuiltin = function(name) {
            return !!name && name.includes("Promise");
        };

        this.builtinEnter = function (name, f, dis, args) {
            if (this.isInterestingBuiltin(name)) {
                console.log("" + dis);
            }
        };
    }

    sandbox.analysis = new MyAnalysis();
})(J$);
```

### Analysis output
```
[e] (*<builtin>:1:1:1:2): unknown exception happened in event handler builtin[Post] NullPointerException
[e] (*<builtin>:1:1:1:2): BUILTIN inputs: null
java.lang.NullPointerException
	at com.oracle.truffle.js.nodes.access.PropertyGetNode.createJavaPropertyNodeMaybe(PropertyGetNode.java:1633)
	at com.oracle.truffle.js.nodes.access.PropertyGetNode.createUndefinedPropertyNode(PropertyGetNode.java:1643)
	at com.oracle.truffle.js.nodes.access.PropertyGetNode.createUndefinedPropertyNode(PropertyGetNode.java:124)
	at com.oracle.truffle.js.nodes.access.PropertyCacheNode.createSpecialization(PropertyCacheNode.java:1259)
	at com.oracle.truffle.js.nodes.access.PropertyCacheNode.specialize(PropertyCacheNode.java:1177)
	at com.oracle.truffle.js.nodes.access.PropertyCacheNode.specialize(PropertyCacheNode.java:1127)
	at com.oracle.truffle.js.nodes.access.PropertyGetNode.getValue(PropertyGetNode.java:199)
	at com.oracle.truffle.js.nodes.access.PropertyNode.executeWithTarget(PropertyNode.java:139)
...
```
(I can provide a full stack trace if that would be helpful.)

I'm fairly certain the `new Promise(...)` is what causes the buggy invocation of `builtinEnter`. The `dis` value provided to `builtinExit` also causes a crash.

### Observations

From my own testing, it seems that when `console.log("" + dis)` happens in the analysis:
1. `dis === undefined` is false
2. `console.log(dis)` prints `[Object: null prototype] {}`
3. `dis.toString()` crashes with:
```
[e] (*<builtin>:1:1:1:2): error happened in event handler builtin[Pre]
[e] TypeError: (intermediate value).toString is not a function
TypeError: (intermediate value).toString is not a function
MyAnalysis.builtinEnter (/root/analysis/promisePrinter.js:94:33)
new Promise (native)
Object.<anonymous> (/root/program/promise1.js:1:70)
Module._compile (internal/modules/cjs/loader.js:936:30)
Object.<anonymous> (internal/modules/cjs/loader.js:947:10)
Module.load (internal/modules/cjs/loader.js:790:32)
Function._load (internal/modules/cjs/loader.js:703:12)
Function.runMain (internal/modules/cjs/loader.js:999:10)
Object.<anonymous> (/root/nodeprof/src/ch.usi.inf.nodeprof/js/jalangi.js:257:19)
Module._compile (internal/modules/cjs/loader.js:936:30)
```

**Comment by alexjordan** [comment]
Hi @mwaldrich, you are seeing a Graal.js-internal object that does not play nice with _normal_ JS code. In fact it's a singleton object representing `[[Construct]]`:
```
  /**
     * Construct token. Passed from the {@code new} node as {@code this} argument to built-in
     * functions to differentiate between a constructor and a normal call (i.e., [[Construct]] and
     * [[Call]] internal methods, see ES5 13.2.1 and 13.2.2). Must not be passed anywhere else.
     */
```

A convenience workaround to avoid this object leaking to user JS, is in #66.

Using `J$.nativeLog(dis)` may give you further clues in these cases and is generally advisable to use inside the builtin callbacks.

**Comment by alexjordan** [comment]
Fixed by #66
